### ISSUE TITLE
[bug] Sensative Strips Guard 700 not responding on versions > 6.0.3

### ISSUE THREAD

**Issue by h-parks** [zwave-js/zwavejs2mqtt#2115]
### Version

**Checklist:**

- [ x ] a developer has told me to come here.
- [ x ] I have checked the troubleshooting section and my problem is **not** described there.
- [ x ] I have read the changelog and my problem is **not** mentioned there.

Build/Run method

- [ x ] Docker
- [ ] PKG
- [ ] Snap package
- [ ] Manually built (git clone - yarn install - yarn run build )

Zwavejs2Mqtt version: 6.2.0
Z-Wave JS version: 8.9.1

### Describe the bug

Sensative Strips Guard 700 window/door sensors unresponsive after installing v6.0.3

### To Reproduce

1. Install any release > 6.0.3
2. Open window/door with sensative strips guard 700
3. Device does not respond

### Expected behavior

Device responds 

[v620__zwavejs2mqtt-store.zip](https://github.com/zwave-js/zwavejs2mqtt/files/7789062/v620__zwavejs2mqtt-store.zip)
[v603__zwavejs2mqtt-store.zip](https://github.com/zwave-js/zwavejs2mqtt/files/7789063/v603__zwavejs2mqtt-store.zip)

**Comment by robertsLando** [comment]
In logs I see lot of

>  2021-12-29T13:16:49.280Z DRIVER « [Node 067] [REQ] [ApplicationCommand]
                                  └─[Security2CCMessageEncapsulation] [INVALID]
                                      error: Security2CC_CannotDecode
2021-12-29T13:16:49.281Z CNTRLR » [Node 067] Message authentication failed, cannot decode command. Requesting a 
                                  nonce...

Did you changed your security S2 keys? 

cc @AlCalzone

**Comment by h-parks** [comment]
No, but I will verify that the keys remain the same between versions.

**Comment by h-parks** [comment]
[attempt_2.zip](https://github.com/zwave-js/zwavejs2mqtt/files/7789105/attempt_2.zip)

Keys stayed the same between versions. Device responds to one "open/close" and then goes unresponsive.

**Comment by AlCalzone** [comment]
I think the main problem is that the device keeps re-sending the same report for minutes. I can't say if this is the one:
```
2021-12-29T13:15:29.321Z DRIVER « [Node 067] [REQ] [ApplicationCommand]
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 195
                                    │ extensions:
                                    │ · type: SPAN
                                    │   sender EI: 0x83d4dd8b1cdee93778c8a314bdaab62c
                                    └─[SupervisionCCGet]
                                      │ session id:      36
                                      │ request updates: false
                                      └─[NotificationCCReport]
                                          notification type:   Access Control
                                          notification status: 255
                                          notification state:  Window/door is open
```
but even minutes later you're having this in the logs:
```
2021-12-29T13:21:08.347Z DRIVER « [Node 067] [REQ] [ApplicationCommand]
                                  └─[Security2CCMessageEncapsulation] [INVALID]
                                      error: Duplicate command
```

Per the S2 specifications, the driver MUST discard duplicate commands (which are detected by looking at the sequence number).

**Comment by h-parks** [comment]
Node 67 is one of about 10 such nodes — I’m curious if the duplicate message is interpreted by the software as _any_ message from the same node with the same attributes — which would explain why the switch changes states once and never more than once on versions greater than 6.0.3.

**Comment by AlCalzone** [comment]
Duplicate message detection is done on a per-node basis. In the case of S2-encapsulated messages, a message is a duplicate if its sequence number (which must be increased for each message) is in the list of the last 10 received sequence numbers from that node.

**Comment by mundschenk-at** [comment]
I find it interesting that I'm having issues with my Sensative Strips Guard 700 as well since first updating to an 8.9.x-beta (accidentally, using the HA addon). I'm not sure it's the same issue, because I don't seem to see those duplicate commands anymore, but it looks like a suspicious amount of coincidence. 

For me it appears that open/close still triggers a battery state update, but no access control change. Reinterviewing affected nodes leaves the access control value as "undefined". Only a local reset seems to such nodes to respond normally again (at least once). 

(If that sounds at all similar, I can elaborate what I tried etc., but I don't want to spam the ticket if it is only a coincidence after all.)

**Comment by AlCalzone** [comment]
Just post a driver log highlighting your issues and we can check.

**Comment by mundschenk-at** [comment]
Sure. Here's today's log. Node 87 is one of the working ones, node 97 a non-working one:
[zwavejs_2022-01-05 3.log](https://github.com/zwave-js/zwavejs2mqtt/files/7815176/zwavejs_2022-01-05.3.log)

I got a new firmware from Sensative (10.28 instead of 8.26), but so far only managed to install it on a single device (not in this log, there's is only one unsuccessful attempt on node 87). In all other cases, I got "Duplicate command" errors early in the process.

**Comment by AlCalzone** [comment]
Node 97 seems to send this notification instead:
- notification state:  Magnetic field interference detected

I'm not sure what the implication of this is, but it could mean that it is unable to detect the open/close status. Maybe an issue mounting the sensor?

**Comment by h-parks** [comment]
New 10.28 Sensative firmware initially appears to resolve the issue based on testing 2 devices in 6.2.0 after updating firmware via 6.0.3. Rolled back to 6.0.3 after initial tests to update the firmware on the rest of the devices. Will confirm once complete.

Update: confirmed working normally.

Magnetic field interference: very possibly a side effect of the wake/pair/etc. process, where a user waves a magnet in front of the sensor a few times to make it wake up.

**Comment by mundschenk-at** [comment]
> Magnetic field interference: very possibly a side effect of the wake/pair/etc. process, where a user waves a magnet in front of the sensor a few times to make it wake up.

Yes, the magnetic interference is from manually waking the node with the magnet (to update its firmware, only to later realize that Z2M hadn't started the process - apparently only every second FW selection really triggers the process).

My current problem is that except for one device, all update attempts fail with lots of "Duplicate command" errors. I can't roll back easily because the node IDs have changed since I upgraded the HA addon. Since I do own a "second" controller now (the 700 series), I might set up a separate Pi with manually installed Z2M - is there a checklist somewhere for manual installation?

**Comment by AlCalzone** [comment]
> all update attempts fail with lots of "Duplicate command" errors

If you're able to, bring the node close to the controller, perform a node heal so it knows the new routes and then do the update.

**Comment by mundschenk-at** [comment]
Unfortunately, that's not really possible as the strips are permanently mounted. This also happens with nodes in the same room (direct line of sight to the controller).

**Comment by AlCalzone** [comment]
I'm going to need some logs of these failed communication attempts then.

**Comment by mundschenk-at** [comment]
Here you are: [zwavejs_2022-01-06 2.log](https://github.com/zwave-js/zwavejs2mqtt/files/7823286/zwavejs_2022-01-06.2.log)

**Comment by AlCalzone** [comment]
The two duplicate commands have the sequence numbers 123 and 124. 123 should have been received shortly before your snippet, 124 is the magnetic sensor report. Both seem to expect a `SupervisionCCReport` in return which have been sent and acknowledged by the node here:
```
2022-01-06T16:20:39.849Z DRIVER » [Node 087] [REQ] [SendData]
                                  │ transmit options: 0x25
                                  │ callback id:      13
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 67
                                    └─[SupervisionCCReport]
                                        session id:          32
                                        more updates follow: false
                                        status:              Success
                                        duration:            0s
2022-01-06T16:20:39.861Z SERIAL « [ACK]                                                                   (0x06)
2022-01-06T16:20:39.865Z SERIAL « 0x0104011301e8                                                       (6 bytes)
2022-01-06T16:20:39.866Z SERIAL » [ACK]                                                                   (0x06)
2022-01-06T16:20:39.868Z DRIVER « [RES] [SendData]
                                    was sent: true
2022-01-06T16:20:39.878Z SERIAL « 0x011800130d00000200ad7f7f7f7f000003000000000301000057              (26 bytes)
2022-01-06T16:20:39.879Z SERIAL » [ACK]                                                                   (0x06)
2022-01-06T16:20:39.881Z DRIVER « [REQ] [SendData]
                                    callback id:            13
                                    transmit status:        OK, took 20 ms
                                    routing attempts:       1
                                    protocol & route speed: Z-Wave, 100 kbit/s
                                    ACK RSSI:               -83 dBm
                                    ACK channel no.:        0
                                    TX channel no.:         0
2022-01-06T16:20:39.901Z SERIAL » 0x0118001357119f03440072f894841825f8a377ee5af719250e90              (26 bytes)
2022-01-06T16:20:39.902Z DRIVER » [Node 087] [REQ] [SendData]
                                  │ transmit options: 0x25
                                  │ callback id:      14
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 68
                                    └─[SupervisionCCReport]
                                        session id:          33
                                        more updates follow: false
                                        status:              Success
                                        duration:            0s
2022-01-06T16:20:39.907Z SERIAL « [ACK]                                                                   (0x06)
2022-01-06T16:20:39.912Z SERIAL « 0x0104011301e8                                                       (6 bytes)
2022-01-06T16:20:39.912Z SERIAL » [ACK]                                                                   (0x06)
2022-01-06T16:20:39.914Z DRIVER « [RES] [SendData]
                                    was sent: true
2022-01-06T16:20:39.935Z SERIAL « 0x011800130e00000200b07f7f7f7f000003000000000301000049              (26 bytes)
2022-01-06T16:20:39.937Z SERIAL » [ACK]                                                                   (0x06)
2022-01-06T16:20:39.939Z DRIVER « [REQ] [SendData]
                                    callback id:            14
                                    transmit status:        OK, took 20 ms
                                    routing attempts:       1
                                    protocol & route speed: Z-Wave, 100 kbit/s
                                    ACK RSSI:               -80 dBm
                                    ACK channel no.:        0
                                    TX channel no.:         0
```

I'm not sure why they are repeated despite being received. 

Also, the sensor does not request any firmware update frames, although it has confirmed the command requesting it to do so:
```
2022-01-06T16:20:40.018Z DRIVER « [Node 087] [REQ] [ApplicationCommand]
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 127
                                    └─[FirmwareUpdateMetaDataCCRequestReport]
                                        status: OK
```
Firmware updates are pulled by the nodes, so if they don't request fragments, there's nothing the driver can do.

You could try re-including the node insecurely and then try the update again - maybe there's something weird in the S2 implementation of that firmware.

**Comment by mundschenk-at** [comment]
Might be a firmware bug? Anyway, something has changed in how the driver interacts with that possibly buggy firmware, because I set up a separate Pi4 with Z2M 6.0.3 (using driver version 8.8.3) and moved the stick to that machine. Lo, and behold: FW updates were finished in half an hour for all remaining strips (whereas I had tried to update them using Z2M 6.2.0/driver 8.9.1 for two days in a row and only managed 2 out of 8 in that time).

(Should we move this discussion to https://github.com/zwave-js/node-zwave-js/issues, BTW?)
